### ISSUE TITLE
Simulator "Report An Issue" System  persistent abuse.

### ISSUE THREAD

**Issue by Malavi1** [CircuitVerse/CircuitVerse#4436]
### Is your feature request related to a problem? Please describe

In the simulator, a few people are misusing the _**Report an issue**_  feature. While reporting they are using abusive/profanity words for no reason. As we overlook some of the **fulfilment** the issue reporter has to qualify before reporting my feature request is to add those. 

* As we make adding email an **optional** choice while reporting an issue it even encourages anonymous people to report whatever they want. On top of that previously the label was Email but it even accepts text/words we have to make sure it has to be a proper email.
* Additionally previous system did not check if the report message had any abusive/profanity words before accepting an issue.

### Describe the solution you'd like

**conditions** 

1. Email - Mandatory
2. Do Not accept the report if it contains any inappropriate words.
 
Before accepting the report from the person who is reporting we have to check the 2 conditions only after the fulfilment of those it will accept otherwise alert the issue reporter.

Previously as we used gem `profanity_filter` in the project description we can use the same logic here in the report message

![sol'n flow](https://github.com/CircuitVerse/CircuitVerse/assets/112646623/93054051-d4a4-4142-942e-c95dee38b0fc)

### Are you working on this?

Yes

**Comment by Malavi1** [comment]
@tachyons can you assign this to me?

**Comment by Malavi1** [comment]
we need to replicate this solution in the forum feature as well.
* The first condition is satisfying (Mandatory Login)
* while commenting/creating questions it not prevent the user from using abusive/profanity words

**Comment by salmoneatenbybear** [comment]
@092vk can we use the profane language checker we were discussing in the weekly meets here ? 

Cc : @tachyons

**Comment by 092vk** [comment]
@salmoneatenbybear yeah , I made it for description and username , we can apply it here also .

**Comment by salmoneatenbybear** [comment]
@092vk okay I have added this for review, if we get a green flag from @tachyons then you can get this fixed by a pr ?
